graph/path/yen_ksp.go

```diff
@@ -0,0 +1,81 @@
+// Copyright ©2015 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package path
+
+import (
+	"reflect"
+	"sort"
+	
+	"gonum.org/v1/gonum/graph"
+)
+
+type YenShortest struct {
+	path []graph.node,
+	weight float64
+}
+
+func YenKSP(source graph.Node, sink graph.Node, g graph.Graph, k int) []Shortest {
+	paths := make([]Shortest, k)
```

Comment:
This doesn't seem like the right thing to do. We don't know that there will be `k` paths, just that there will be at most `k` paths. The WP-published algorithm does this (sort of), but it is horrifically inefficient as an approach to returning `k` paths from s to t. The Dijkstra that is being used in the article is just returning the shortest path from s to t, while we return all shortest paths from s. What should happen here is that you reuse the `Shortest` you have to recalculate the shortest path after the graph pseudomutation and then keep the actual path (a `[]graph.Node` is fine for this) and finally return a `[][]graph.Node`.